[UX] Disabling tabs (#209)
mautic/developer-documentation-new: docs/design/disabling_elements.rst

Reviewer: See comment on heading nesting, this is using the wrong underlines.

```
H1 ###
H2 ***
H3 ===
H4 ---
H5 ~~~
```

> Disabling interface elements

Reviewer: ```suggestion
Making interface elements inactive or unavailable
```

Let's not use disabling and instead talk about what we're actually doing - making something inactive or unavailable.

Author: We're doing this: making users become not allowed to interact with that thing, effectively removing their ability of taking actions

in the code, we use:
`cursor: not-allowed;`
to show the red X cursor as a feedback

another way to say is that we're removing pointer events, as this attribute is also appended to ensure users have their most deep desire and wishes to interact completely blocked:
`pointer-events: none;`

how would you find better to call?

Author: it's great to pick 1 word to avoid having the synonyms (inactive or unavailable) there in every use

=======================================

> Disabling interface elements is a crucial aspect of user interface design, providing visual feedback and preventing user interaction when certain actions are not allowed.

Reviewer: ```suggestion
The state of interface elements is a crucial aspect of user interface design, providing visual feedback and preventing interaction when certain actions aren't allowed.
```

> Disabling tabs

Reviewer: ```suggestion
Changing tab states
```

--------------

> We use the following CSS code to style disabled tabs:

Reviewer: ```suggestion
Use the following CSS code to style inactive or unavailable tabs:
```

.. code-block:: css

   .nav-tabs.nav-tabs-contained > li.disabled {
     cursor: not-allowed;
     color: var(--text-disabled);
   }

   .nav-tabs.nav-tabs-contained > li.disabled > a {
     background-color: var(--button-disabled);
     pointer-events: none;
   }

This CSS accomplishes the following:

> * Sets the cursor to ``not-allowed`` for disabled tabs, indicating that interaction is prohibited.

Reviewer: ```suggestion
* Sets the cursor to ``not-allowed`` for inactive or unavailable tabs, indicating that interaction is prohibited.
```

> * Changes the text color to a predefined disabled state color.

Reviewer: ```suggestion
* Changes the text color to a predefined inactive or unavailable state color.
```

> * Modifies the background color of the tab to visually represent its disabled state.

Reviewer: ```suggestion
* Modifies the background color of the tab to visually represent its inactive or unavailable state.
```

> * Disables click events on the tab using ``pointer-events: none``.

Reviewer: ```suggestion
* Prevent click events on the tab using ``pointer-events: none``.
```

> To dynamically disable tabs, we use JavaScript to add or remove the ``disabled`` class. Here's an example function:

Reviewer: ```suggestion
To dynamically deactivate tabs, use JavaScript to add or remove the ``disabled`` class. Here's an example function:
```

.. code-block:: javascript

   Mautic.togglePermissionVisibility = function () {
       setTimeout(function () {
           if (mQuery('#role_isAdmin_0').prop('checked')) {
               mQuery('#permissions-tab').removeClass('disabled');
           } else {
               mQuery('#permissions-tab').addClass('disabled');
           }
       }, 10);
   };

This function:

> * Checks the state of a checkbox (``#role_isAdmin_0``).

Reviewer: ```suggestion
* Checks the state of a checkbox - ``#role_isAdmin_0``.
```

* Adds or removes the ``disabled`` class from the permissions tab based on the checkbox state.

> To implement disabled states for tabs:

Reviewer: ```suggestion
To implement inactive or unavailable states for tabs:
```

1. Assign unique IDs to your tab elements.
2. Use JavaScript to toggle the ``disabled`` class on the appropriate ``<li>`` elements.

.. note::

>    Always use disabled states instead of hiding elements.

Reviewer: ```suggestion
   Always use inactive or unavailable states instead of hiding elements.
```
